app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java

```diff
@@ -0,0 +1,57 @@
+package org.pesmypetcare.mypetcare.controllers.community;
+
+import org.pesmypetcare.mypetcare.features.community.forums.NotForumOwnerException;
+import org.pesmypetcare.mypetcare.features.community.posts.Post;
+import org.pesmypetcare.mypetcare.features.users.User;
+import org.pesmypetcare.mypetcare.services.community.CommunityService;
+
+/**
+ * @author Xavier Campos
+ */
+public class TrUnbanPost {
+    private CommunityService communityService;
+    private User user;
+    private Post post;
+    private boolean result;
+
+    public TrUnbanPost(CommunityService communityService) {
+        this.communityService = communityService;
+    }
+
+    /**
+     * Setter of the user that wants to unban the post.
+     * @param user
+     */
+    public void setUser(User user) {
+        this.user = user;
+    }
+
+    /**
+     * Setter of the post that has to be unbanned
+     * @param post
+     */
+    public void setPost(Post post) {
+        this.post = post;
+    }
+
+    /**
+     * Returns the result of the transaction.
+     * @return True if the transaction was successful or false otherwise
+     */
+    public boolean isResult() {
+        return result;
+    }
+
+    /**
+     * Executes the transaction.
+     * @throws NotForumOwnerException The given user is not the owner of the forum were the post is made
+     */
+    public void execute() throws NotForumOwnerException {
+        if (post.getForum().getOwnerUsername().equals(user.getUsername())) {
+            this.communityService.unbanPost(user, post);
+        }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['}' at column 9 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)